scylla/src/lib.rs

```diff
@@ -257,3 +257,22 @@ pub(crate) mod utils;
 
 #[cfg(test)]
 pub(crate) use utils::test_utils;
+
+#[cfg(feature = "unstable-testing")]
+pub mod internal_testing {
+    use scylla_cql::serialize::row::SerializedValues;
+
+    use crate::routing::partitioner::PartitionerName;
+    use crate::routing::Token;
+    use crate::statement::prepared::TokenCalculationError;
+
+    pub fn calculate_token_for_partition_key(
+        serialized_partition_key_values: &SerializedValues,
+        partitioner: &PartitionerName,
+    ) -> Result<Token, TokenCalculationError> {
+        crate::routing::partitioner::calculate_token_for_partition_key(
+            serialized_partition_key_values,
+            partitioner,
+        )
+    }
+}
```

Comment:
:question: Once we have this, can we deduplicate some test utilities that used to be present for both `scylla` unit tests and integration tests?

Reply:
Theoretically we could deduplicate them by only having them in the lib (instead of having them in integration tests) under this feature, like we have `calculate_token_for_partition_key` now.
This is a tradeoff: running integration tests would require passing a flag to `cargo test` to enable this feature, otherwise the integration tests target will not exist at all.
It's not a problem for CI, nor when using Makefile, but it is making it less convenient to call `cargo test` directly.

For benchmark doing this seemed ok to me (because of how rarely we run them). For integration tests I'm not sure.